server/src/services/AccountService.js

```diff
@@ -181,6 +175,82 @@ class AccountService {
 
     return { status: 200, success: true, data };
   }
+
+  /**
+   *
+   * Handles the logic for fetching the details of a single account
+   * @static
+   * @param {number} accountNumber
+   * @returns {object} API Response Object
+   * @memberof AccountService
+   */
+  static async fetchOneAccount(accountNumber, user, userType) {
+    const notAllowed = await AccountService.checkPermission(
+      accountNumber,
+      user,
+      userType,
+      'account'
+    );
+    if (notAllowed) return notAllowed;
+
+    const foundAccount = await Account.findAccount(accountNumber);
+    const { email, status, type, balance, createdon } = foundAccount;
+    return {
+      status: 200,
+      success: true,
+      data: {
+        createdOn: createdon,
+        ownerEmail: email,
+        accountNumber,
+        status,
+        type,
+        balance
+      }
+    };
+  }
+
+  /**
+   *
+   * Restricts access of specified route to account owners and staff users
+   * @static
+   * @param {number} accountNumber
+   * @param {number} user user id
+   * @param {string} userType currently logged in user type(client or staff)
+   * @param {string} route route to guard
+   * @returns {object|boolean} error object or boolean if check passes
+   * @memberof AccountService
+   */
+  static async checkPermission(accountNumber, user, userType, route) {
```

Comment:
Why are you doing this check here, I suggest you move it to a middleware where you do your validations.
If there is any specific reason why you have to do the check here let me know

Reply:
> Why are you doing this check here, I suggest you move it to a middleware where you do your validations.
> If there is any specific reason why you have to do the check here let me know

I thought about moving it to a middleware, but i'm also planning to use it where i wrote something similar in a function in my transaction service, in the transaction service i was making a query to the database before i could get the account number, which is quite different to how i'm getting the account number above, i'm getting the account number from the incoming request above, but i plan to refactor it to a helper function, so that i can use it across the three places with ease